synapse/handlers/presence.py

```diff
@@ -331,8 +331,8 @@ async def add_users_to_send_full_presence_to(self, user_ids: Iterable[str]):
         # correctly won't receive a second snapshot.
 
         # Get the current presence state for one of the users (defaults to offline if not found)
-        user_id = yield user_ids
-        current_presence_state = await self.get_state(user_id)
+        user_id = next(iter(user_ids))
```

Comment:
Actually, this doesn't make sense, since we've already consumed the iterable (some iterables can only be iterated over once). I think we should change the function definition to be a `Collection` and then you can do `[0]`? We probably also want a empty list check.

Reply:
Oh you're right, that would definitely be a difficult one to debug if we were always pulling the first item of the list out, instead of just checking it.

Reply:
Although apparently `Collection` doesn't define `__getitem__` either.

Reply:
A `Sequence` may work...

Reply:
Eh, `next(iter(user_ids))` is fine if we've already done an empty check.

Reply:
Have added an empty check just to be sure.